OOP/oop_home_task.py

```diff
@@ -0,0 +1,475 @@
+        """
+        Writes expiration date of publication to file
+        @param publications_file: opened file object
+        @param expiration_date: date of publication
+        """
+        publications_file.write(f'Actual until: {expiration_date}, ')
+
+    @staticmethod
+    def write_expiration_days(publications_file, expiration_days):
+        """
+        Writes expiration days of publication to file
+        @param publications_file: opened file object
+        @param expiration_days: date of publication
+        """
+        publications_file.write(f'{expiration_days} days left\n')
+
+    @staticmethod
+    def write_author(publications_file, author):
+        """
+        Writes author of publication to file
+        @param publications_file: opened file object
+        @param author: author of publication
+        """
+        publications_file.write(f'Text author: {author}. ')
+
+    @staticmethod
+    def write_text_total_symbols(publications_file, symbols_count):
+        """
+        Writes count of symbols in publication text to file
+        @param publications_file: opened file object
+        @param symbols_count: count of symbols
+        """
+        publications_file.write(f'Symbols count: {symbols_count}.\n')
+
+    @staticmethod
+    def write_ending(publications_file, separator):
+        """
+        Writes publications separator to file
+        @param publications_file: opened file object
+        @param separator: publications SEPARATOR
+        """
+        publications_file.write(f'{separator}')
+        pass
+
+    @staticmethod
+    def close_file(publications_file):
+        """
+        Closes publications file
+        @param publications_file: opened file object
+        """
+        publications_file.close()
+
+
+class Application(News, Advertisement, Article):
+    """
+    Class Application that inherits News, Advertisement, Article classes.
+    Used for creation of final publication scenarios and running of application.
+    """
+    def __init__(self, city='', publication_text='', publication_date=''):
+        """
+        Method is constructor for Application class
+        @param publication_text: text of publication, empty by default
+        @param publication_date: date of publication, empty by default
+        @param city: city where something happened, empty by default
+        """
+        super().__init__(publication_date, city, publication_text)
+
+    def publish_news(self):
+        """
+        Method for publishing of news scenario.
+        Method creates objects of UserInterface, FileWriter, ObjectCreator classes.
+        Asks user to enter publication text, city.
+        After writes entered data to file in turn using FileWriter methods.
+        Catches exception if target file not found.
+        """
+        try:
+            user_interface = UserInterface()
+            file_writer = FileWriter()
+            file = file_writer.open_file('News_file.txt')
```

Comment:
As you open it inside method what will happen if your program break before file closing?
Usually for opening and operating with files, streams, connections etc context managers are used. It's the beast practice here.

Reply:
Context managers added.